### ISSUE TITLE
sveltekit getting-started docs not working

### ISSUE THREAD

**Issue by engageintellect** [vercel-labs/ai#88]
I am using the directions from here: https://sdk.vercel.ai/docs/getting-started#prerequisites

Following everything as explained, setting my OPENAI_API_KEY in .env, creating a /routes/api/completion.js file and pasting the associated code,  and keep getting the following error when clicking the "generate slogan" button in my /src/routes/index.svelte (or /src/routes/+page.svelte)

```
Error: Not found: /api/completion
```

Am I missing something here?

**Comment by anthonyko** [comment]
Try renaming the `/routes/api/completion.js ` file to `/routes/api/completion/+server.js`

**Comment by AdrianFerrer1** [comment]
I'm getting "Failed to convert the response to stream. Received status code: 429." on visual studio code console and "POST http://localhost:5173/api/completion 500 (Internal Server Error)" on the browser's console

**Comment by engageintellect** [comment]
> I'm getting "Failed to convert the response to stream. Received status code: 429." on visual studio code console and "POST http://localhost:5173/api/completion 500 (Internal Server Error)" on the browser's console 

Deploy your current code to vercel. It should work. I too get that error when running on dev... still haven't been able to solve for that or get any responses on dev.

**Comment by engageintellect** [comment]
> Try renaming the `/routes/api/completion.js ` file to `/routes/api/completion/+server.js` and rename `src/routes/index.svelte` to `src/routes/+page.svelte`.
> 
> 
> 
> and I think the POST needs to be capitalized.
> 
> ```
> 
> export async function POST({ request }) {
> 
> 	const { prompt } = await request.json();
> 
> 
> 
> ...
> 
> }
> 
> ```
> 
> 
> 
> This should at least get a response displaying for you. However, it's not completely working for me. The response is incomplete and the stream isn't finishing for me.
> 
> 
> 
> <img width="356" alt="CleanShot 2023-06-15 at 21 24 57@2x" src="https://github.com/vercel-labs/ai/assets/4150464/4c06e4da-0227-41ef-b753-8931a658e2b7">
> 
> 

That's odd... this was one of the first things I tried before posting this issue and I wasn't able to get anything to work (on dev)

Will have to try again and see what I was missing.


Are you able to get a response locally? Or only once deployed?

**Comment by anthonyko** [comment]
> Are you able to get a response locally? Or only once deployed?

I am running it all locally. I haven't bothered deploying yet. There is updated code in the [Getting started section](https://sdk.vercel.ai/docs/getting-started). Give that a shot. It was working when I copy pasted just now.

I still don't get the completed stream though.

**Comment by engageintellect** [comment]
> > Are you able to get a response locally? Or only once deployed?
> 
> 
> 
> I am running it all locally. I haven't bothered deploying yet. There is updated code in the [Getting started section](https://sdk.vercel.ai/docs/getting-started). Give that a shot. It was working when I copy pasted just now.
> 
> 
> 
> I still don't get the completed stream though. 

Thanks for this update. I will try the updated code.

Please do let me know if you get a completed stream once deployed. I'm assuming it will work.

**Comment by AdrianFerrer1** [comment]
> > I'm getting "Failed to convert the response to stream. Received status code: 429." on visual studio code console and "POST http://localhost:5173/api/completion 500 (Internal Server Error)" on the browser's console
> 
> Deploy your current code to vercel. It should work. I too get that error when running on dev... still haven't been able to solve for that or get any responses on dev.

And is there a way to not run it on dev? because having to deploy the app everytime I want to check something sounds like too much

**Comment by AdrianFerrer1** [comment]
> > I'm getting "Failed to convert the response to stream. Received status code: 429." on visual studio code console and "POST http://localhost:5173/api/completion 500 (Internal Server Error)" on the browser's console
> 
> Deploy your current code to vercel. It should work. I too get that error when running on dev... still haven't been able to solve for that or get any responses on dev.

Just deployed on vercel, I'm still getting the same error

**Comment by engageintellect** [comment]
> > > I'm getting "Failed to convert the response to stream. Received status code: 429." on visual studio code console and "POST http://localhost:5173/api/completion 500 (Internal Server Error)" on the browser's console
> 
> > 
> 
> > Deploy your current code to vercel. It should work. I too get that error when running on dev... still haven't been able to solve for that or get any responses on dev.
> 
> 
> 
> And is there a way to not run it on dev? because having to deploy the app everytime I want to check something sounds like too much

This is my current issue as well.

**Comment by paolodina** [comment]
@anthonyko for the incomplete response, I got it work specifying a value for `max_token`. HTH

```javascript
const response = await openai.createCompletion({
    model: 'text-davinci-003',
    ...
    max_tokens: 100,
    ...
})
```

**Comment by engageintellect** [comment]
> @anthonyko for the incomplete response, I got it work specifying a value for `max_token`. HTH
> 
> 
> 
> ```javascript
> 
> const response = await openai.createCompletion({
> 
>     model: 'text-davinci-003',
> 
>     ...
> 
>     max_tokens: 100,
> 
>     ...
> 
> })
> 
> ```

Is this on a dev server?

Also, any idea what the max tokens should be set at? Or how many are used for a typical response?

**Comment by anthonyko** [comment]
Ah thank you! I assumed it defaulted to infinite max tokens, but it looks like chat and completions are different. 

Completions [default to 16 tokens](https://platform.openai.com/docs/api-reference/completions/create#completions/create-max_tokens) while [chat defaults to infinite tokens](https://platform.openai.com/docs/api-reference/chat/create#chat/create-max_tokens).

**Comment by anthonyko** [comment]
> Is this on a dev server?

yes, dev server.

They have a [token counter](https://platform.openai.com/tokenizer) that gives you an idea of what is being used.

**Comment by AdrianFerrer1** [comment]
> ```js
> max_tokens: 100,
> ```

That still doesn't solve the issue we have, it just solves the issue @anthonyko was having on his recreation

**Comment by engageintellect** [comment]
> > ```js
> 
> > max_tokens: 100,
> 
> > ```
> 
> 
> 
> That still doesn't solve the issue we have, it just solves the issue @anthonyko was having on his recreation

Agreed

**Comment by paolodina** [comment]
@AdrianFerrer1 @engageintellect 

https://github.com/vercel-labs/ai/issues/144#issuecomment-1596158004 (please try this, worked for me)

**Comment by AdrianFerrer1** [comment]
> @AdrianFerrer1 @engageintellect
> 
> [#144 (comment)](https://github.com/vercel-labs/ai/issues/144#issuecomment-1596158004) (please try this, worked for me)

![image](https://github.com/vercel-labs/ai/assets/112191768/28ce394f-c3f1-4fca-aa59-868ccb7cefbb)

I have created a new issue with the code that I'm using, I may have done something wrong most probably #152

**Comment by AdrianFerrer1** [comment]
> > > 
> > 
> > 
> > > max_tokens: 100,
> > 
> > 
> > > 
> > 
> > 
> > That still doesn't solve the issue we have, it just solves the issue @anthonyko was having on his recreation
> 
> Agreed

Update on this. You HAVE TO put a card on the biling info section and u need to have at least 5 usd btw

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Documentation**  
Confidence: **High**

### Rationale
The primary issue is incorrect or outdated SvelteKit instructions in the Getting Started documentation, requiring a documentation correction rather than a new SDK capability.

### Evidence
- The Getting Started guide instructed SvelteKit users to create `/routes/api/completion.js`, resulting in a 404 for `/api/completion`.
- A commenter identified the correct SvelteKit route as `/routes/api/completion/+server.js` with an uppercase `POST` handler.
- A later comment states that the Getting Started code was updated and then worked when copied locally.
